homeassistant/components/squeezebox/coordinator.py

```diff
@@ -0,0 +1,84 @@
+"""DataUpdateCoordinator for the Squeezebox integration."""
+
+from asyncio import timeout
+from datetime import timedelta
+import logging
+import re
+
+from pysqueezebox import Server
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    SENSOR_UPDATE_INTERVAL,
+    STATUS_API_TIMEOUT,
+    STATUS_SENSOR_LASTSCAN,
+    STATUS_SENSOR_NEEDSRESTART,
+    STATUS_SENSOR_NEWPLUGINS,
+    STATUS_SENSOR_NEWVERSION,
+    STATUS_SENSOR_RESCAN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LMSStatusDataUpdateCoordinator(DataUpdateCoordinator):
+    """LMS Status custom coordinator."""
+
+    def __init__(self, hass: HomeAssistant, my_api: Server) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=my_api.name,
+            update_interval=timedelta(seconds=SENSOR_UPDATE_INTERVAL),
+            always_update=False,
+        )
+        self.my_api = my_api
+        self.newversion_regex = re.compile("<.*$")
+
+    async def _async_update_data(self):
+        """Fetch data fromn LMS status call.
+
+        Then we process only a subset to make then nice for HA
+        """
+        # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+        # handled by the data update coordinator.
+        async with timeout(STATUS_API_TIMEOUT):
+            data = await self.my_api.async_status()
+
+        if not data:
+            raise UpdateFailed("No data from status poll")
+        _LOGGER.debug("Raw serverstatus %s=%s", self.my_api.name, data)
+
+        return self._prepare_status_data(data)
+
+    def _prepare_status_data(self, data: dict) -> dict:
+        """Sensors that need the data changing for HA presentation."""
+
+        # 'lastscan': '1718431678', epoc -> ISO 8601 not always present
+        data[STATUS_SENSOR_LASTSCAN] = (
+            STATUS_SENSOR_LASTSCAN in data
+            and dt_util.utc_from_timestamp(int(data[STATUS_SENSOR_LASTSCAN]))
+            or None
+        )
+        # rescan bool are we rescanning alter poll not always present
+        data[STATUS_SENSOR_RESCAN] = STATUS_SENSOR_RESCAN in data and True or False
+        # needsrestart bool plugin updates... not always present
+        data[STATUS_SENSOR_NEEDSRESTART] = (
```

Comment:
With lastscan and needsrestart, since we can control how we put the data in the coordinator data, can't we rename them to `last_scan` and `needs_restart`

Reply:
Make the CONTS more readable? we could but I think its clearer to use the LMS term and present it better for the user via the translation as your have suggested.